### PR TITLE
[Merged by Bors] - feat: port Analysis.InnerProductSpace.Orientation

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by Parcly-Taxel** [comment]
Note the [out of sync queue](https://leanprover-community.github.io/mathlib-port-status/file/analysis/inner_product_space/orientation) for this file.

File: Mathlib/Analysis/InnerProductSpace/Orientation.lean [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+        AlternatingMap.constOfIsEmpty_apply, Function.const_apply, abs_one, Finset.univ_eq_empty, Finset.prod_empty,
+        le_refl]
+    · rintro rfl
+      simp only [Nat.zero_eq, volumeForm_neg_orientation, volumeForm_zero_pos,
+        AlternatingMap.constLinearEquivOfIsEmpty_apply, AlternatingMap.neg_apply, AlternatingMap.constOfIsEmpty_apply,
+        Function.const_apply, abs_neg, abs_one, Finset.univ_eq_empty, Finset.prod_empty, le_refl]
+  haveI : FiniteDimensional ℝ E := fact_finiteDimensional_of_finrank_eq_succ n
+  have : finrank ℝ E = Fintype.card (Fin n.succ) := by simpa using _i.out
+  let b : OrthonormalBasis (Fin n.succ) ℝ E := gramSchmidtOrthonormalBasis this v
+  have hb : b.toBasis.det v = ∏ i, ⟪b i, v i⟫ := gramSchmidtOrthonormalBasis_det this v
```
Comment (Ruben-VandeVelde) [inline Mathlib/Analysis/InnerProductSpace/Orientation.lean:285]: So this is fun. `volumeForm_robust'` is a statement about `@Basis.det .. (fun a b ↦ instDecidableEqFin n a b)` (https://leanprover-community.github.io/mathlib4_docs/Init/Prelude.html#instDecidableEqFin), while `gramSchmidtOrthonormalBasis_det` is a statement about `@Basis.det  .. (fun a b ↦ instDecidableEq a b)` (https://leanprover-community.github.io/mathlib4_docs/Mathlib/Init/Algebra/Order.html#instDecidableEq)

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Comment by mo271** [comment]
There is still some TODOs, namely deriving some instances that couldn't be inferred...

**Comment by Parcly-Taxel** [comment]
> There is still some TODOs, namely deriving some instances that couldn't be inferred...

I found out that the last `FiniteDimensional \R E` instance can be synthed for the last theorem in the same manner as it is in earlier theorems in the file. I'll push shortly.

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Comment by jcommelin** [comment]
Let's try again :smile: 

bors merge

**Pull request closed without merging**